front/src/applications/operationalStudies/components/Project/PictureUploader.tsx

```diff
@@ -85,106 +74,91 @@ function PicturePlaceholder({ image, isValid, tempProjectImage }: PicturePlaceho
     return <img src={URL.createObjectURL(projectImage)} alt="Project illustration" />;
   }
   return <>{displayNoImageMessages(isValid, t)}</>;
-}
+};
+
+type Categories = {
+  [key: string]: {
+    category_image: string;
+    images: string[];
+  };
+};
 
-function PicturePlaceholderButtons({
+const PicturePlaceholderButtons = ({
   setTempProjectImage,
   tempProjectImage,
   safeWord,
   t,
-}: PicturePlaceholderButtonsProps) {
-  async function getRandomImage(url: string) {
+}: PicturePlaceholderButtonsProps) => {
+  const [categories, setCategories] = useState<Categories>({});
+
+  async function getRandomImage(images: string[]) {
```

Comment:
I wouldn't have chosen a random image with each click.
The algorithm I have in mind is:
1. When the category is loaded pick a random number
2. When the user click, add one to the previous number (modulo the numbe of images)

This way when you have the same image twice it means that you saw all of them.